src/Authenticator/AuthenticatorCollection.php

```diff
@@ -65,12 +65,12 @@ protected function _create($className, $alias, $config)
     /**
      * Resolves authenticator class name.
      *
-     * @param string $class Class name to be resolbed.
-     * @return string
+     * @param string $class Class name to be resolved.
+     * @return string|null
      */
     protected function _resolveClassName($class)
     {
-        return App::className($class, 'Authenticator', 'Authenticator');
+        return (string)App::className($class, 'Authenticator', 'Authenticator') ?: null;
```

Comment:
@robertpustulka Technically phpstan is right since bool means it could also be `true` though our method only return `false`. I would suggest reverting back to level 4 instead of adding unnecessary casting :)